### DISCUSSION TITLE
v0.6.0

### DISCUSSION THREAD

**Discussion by luolongfei** [luolongfei/freenom#246] [Announcements]
- 通过语音识别，自动解决 aws waf 验证
- 优化自动重试逻辑
- 引入 https://hub.docker.com/r/luolongfei/aws-waf-solver ，欢迎 star

基于语音识别的方案实际上我几个月前就提过，但是一直没时间写，最近抽空写了一版，目前属于 beta 版本，可能有各种问题或者情况，欢迎反馈。部署文档请参考：https://github.com/luolongfei/freenom?tab=readme-ov-file#-%E9%80%9A%E8%BF%87-docker-compose-%E9%83%A8%E7%BD%B2

目前只写了 docker compose 的部署文档，其它平台或部署方式等后续补充，或者你也可以写相关文档提 PR 参与到项目中来。

简单说下实现方式：
php 端每次请求前会请 aws-waf-solver 端给下 aws waf token，aws-waf-solver 接到指令后会创建一个异步任务，并返回一个 task id 给 php，这时候，php 会一直轮询 aws-waf-solver：好了没，拿到了吗。aws-waf-solver 会模拟正常人类的方式去访问 freenom 的页面，被人机验证拦住后，转到语音验证，把语音文件丢给 wit.ai 识别，识别成功后从 cookie 中提取 aws waf token，并告诉 php，php 拿到 token 后，再去请求 freenom，如果 token 失效了，php 会自动再找 aws-waf-solver 要。

语音识别有一定的准确率，干扰和 ai 本身的模型都会有影响，我自己服务器测试一般在 2 次内识别成功，但也有群友反馈 5 次内成功的。

这次本来打算把 OpenAI whisper（离线版）也作为配置项加入的，但是群友测试后发现，大模型的尺寸比较大，中等大小的模型可以占用 7G 大小的磁盘，效果上和 fb 的差不了太多，所以暂时移除了相关内容。目前 wit.ai 的效果我基本能接受。

如果本项目有帮到你，欢迎点击下方地址赞助支持，你的每一分钱，都是在为你想要的世界投票：

https://images.llfapp.com/pay.png

https://www.paypal.me/mybsdc

<hr /><em>This discussion was created from the release <a href='https://github.com/luolongfei/freenom/releases/tag/v0.6.0'>v0.6.0</a>.</em>

**Comment by spacex-3** [comment]
latest还是0.5版本，拉取0.6出错

**Reply by luolongfei** [reply]
freenom 的镜像目前没有更新，0.6 是通过 docker compose 实时构建。docker hub 的镜像等我共享 token 版本出来再发布

**Comment by sdwflizheng** [comment]
提示 (Domain name not found) ：
getAwsWafToken error: cURL error 6: Could not resolve: aws-waf-solver (Domain name not found) (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://aws-waf-solver:2019/aws_waf_token

**Reply by luolongfei** [reply]
你这一看就没有用 docker compose 部署

**Reply by sdwflizheng** [reply]
PHP下载文件覆盖的原文件，正在按说明用docker试试^_^

**Comment by jinwyp** [comment]
docker hub 上的镜像还没升级到最新的v0.6.0 ? 还是老版本v0.5.4  ？ 
docker pull luolongfei/freenom  运行后 logs 显示当前程序版本 v0.5.4
